azerothcore/azerothcore-wotlk
# feat(Core/Creature): implement c_t->spell_school_immune_mask

* https://github.com/TrinityCore/TrinityCore/commit/ae878e18ef121b7b1960f40f2b0f99debd2155d8

Co-Authored-By: Wyreth <32145860+Wyreth@users.noreply.github.com>





## Changes Proposed:
-  Implement spell_school_immune_mask to allow defining creature spell school immunity in the database
- Port all existing spell_school_immune_mask from Trinitycore

## Issues Addressed:
- Closes 



## Tests Performed:
- Builds
- Tested in-game on all spell schools as shown below
![image](https://user-images.githubusercontent.com/24550914/106032062-4874a480-60d0-11eb-8761-0f1e289a6e7e.png)
![image](https://user-images.githubusercontent.com/24550914/106032151-680bcd00-60d0-11eb-8793-984c4db82583.png)
![image](https://user-images.githubusercontent.com/24550914/106032261-8b367c80-60d0-11eb-8400-bc4493ee215f.png)
![image](https://user-images.githubusercontent.com/24550914/106032331-a43f2d80-60d0-11eb-93bb-0e59f8e35abc.png)





## How to Test the Changes:

```cpp
    SPELL_SCHOOL_NORMAL                 = 0, (1 >> 0 = 1)
    SPELL_SCHOOL_HOLY                   = 1, (1 >> 1 = 2)
    SPELL_SCHOOL_FIRE                   = 2, (1 >> 2 = 4)
    SPELL_SCHOOL_NATURE                 = 3, (1 >> 3 = 8)
    SPELL_SCHOOL_FROST                  = 4, (1 >> 4 = 16)
    SPELL_SCHOOL_SHADOW                 = 5, (1 >> 5 = 32)
    SPELL_SCHOOL_ARCANE                 = 6  (1 >> 6 = 64)
```

Easily tested with this:
```sql
-- REMEMBER TO UNDO THIS AFTER TESTING AS THIS MOB SHOULD NOT BE IMMUNE TO THIS
UPDATE `creature_template` SET `spell_school_immune_mask`=`spell_school_immune_mask`|1|2|4|8|16|32|64 WHERE `entry` = 299;
```
1. .go c id 299
2. Test casting spells with the different schools. (HOLY, FIRE, NATURE, FROST, SHADOW, ARCANE | NORMAL is Physical damage)

## Known Issues and TODO List:



## Target Branch(es):








 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
